rule/regex.go

```diff
@@ -0,0 +1,36 @@
+package rule
+
+import (
+	"regexp"
+
+	"github.com/confetti-framework/support"
+)
+
+type Regex struct {
+	Expression *regexp.Regexp
+	Error      error
```

Comment:
```suggestion
	err        error
```
The error can remain private (just like you did with InArray).

Reply:
Solved on commit [124b39f](https://github.com/rafaelbreno/validation/commit/124b39f9975963d837eea7209e8a48eb7bae5fb9)